src/main/scala/com/fulcrumgenomics/sv/tools/SvPileup.scala

```diff
@@ -121,157 +120,166 @@ class SvPileup
-      else                                      (maxWithinReadDistance, SplitReadIntraContig)
-    }
-
-    val innerDistance = {
-      if (seg1.range.start <= seg2.range.start) seg2.range.start - seg1.range.end
-      else seg1.range.start - seg2.range.end
-    }
-
-    if (innerDistance <= maxDistance) None
-    else Some(BreakpointEvidence(from=seg1, into=seg2, evidence=evidence))
-  }
-
-  /** Determines if the primary mappings are from a read pair are evidence of an inversion or other re-arrangement from
-   * an unexpected (i.e. not FR) read pair orientation.  This only makes sense if
-   * the segments are from the R1 and R2 primary alignments.
-   *
-   * @param seg1 the first alignment segment
-   * @param seg2 the second alignment segment
-   */
-  def determineOddPairOrientation(seg1: AlignedSegment, seg2: AlignedSegment): Option[BreakpointEvidence] = {
+  def isIntraContigBreakpoint(seg1: AlignedSegment,
+                              seg2: AlignedSegment,
+                              maxWithinReadDistance: Int,
+                              maxBetweenReadDistance: Int): Boolean = {
     require(seg1.range.refIndex == seg2.range.refIndex)
 
-    if (seg1.origin.isPairedWith(seg2.origin)) { // Treat as R1->R2 or R2->R1
-      if (seg1.positiveStrand == seg2.positiveStrand) Some(BreakpointEvidence(from=seg1, into=seg2, evidence=ReadPairTandem))
-      else {
-        // Follows the implementation in htsjdk.samtool.SamPairUtil.getPairOrientation
-        val (positiveStrandFivePrime, negativeStrandFivePrime) = {
-          if (seg1.positiveStrand) (seg1.range.start, seg2.range.end) else (seg2.range.start, seg1.range.end)
-        }
-        if (positiveStrandFivePrime < negativeStrandFivePrime) None // FR is ok
-        else Some(BreakpointEvidence(from=seg1, into=seg2, evidence=ReadPairReverseForward)) // RF
+    // The way aligned segments are generated for a template, if we have all the reads in the expected orientation
+    // the segments should all come out on the same strand.  Therefore any difference in strand is odd.  In addition
+    // any segment that "moves backwards" down the genome is odd, as genome position and read position should increase
+    // together.
+    seg1.positiveStrand != seg2.positiveStrand ||
+      (seg1.positiveStrand && seg2.range.start < seg1.range.end) ||
+      (!seg1.positiveStrand && seg1.range.start < seg2.range.start) || {
+      val maxDistance = if (seg1.origin.isInterRead(seg2.origin)) maxBetweenReadDistance else maxWithinReadDistance
+
+      val innerDistance = {
+        if (seg1.range.start <= seg2.range.start) seg2.range.start - seg1.range.end
+        else                                      seg1.range.start - seg2.range.end
       }
-    }
-    else {
-      if (seg1.positiveStrand == seg2.positiveStrand) None
-      else Some(BreakpointEvidence(from=seg1, into=seg2, evidence=SplitReadOppositeStrand))
+
+      innerDistance > maxDistance
```

Comment:
I don't find the above easier to read, due to mixing of the nested conditionals all being part of the return expression.  Why not the following?  It makes commenting easier

```scala
if (seg1.positiveStrand != seg2.positiveStrand) true // opposite strands
else if (seg1.positiveStrand && seg2.range.start < seg1.range.end) true // one segment before the other
else if (!seg1.positiveStrand && seg1.range.start < seg2.range.end) true // one segment before the other
else {  // check inner distance
  val innerDistance = {
    if (seg1.range.start <= seg2.range.start) seg2.range.start - seg1.range.end
    else                                      seg1.range.start - seg2.range.end
  }
  innerDistance > maxDistance
}
```

Reply:
Fine 😝